surface/include/pcl/surface/impl/convex_hull.hpp

```diff
@@ -182,106 +166,88 @@ pcl::ConvexHull<PointInT>::performReconstruction2D (PointCloud &hull, std::vecto
   }
```

Comment:
Replace all of the above for
```cpp
if (!xy_proj_safe && !yz_proj_safe && !xz_proj_safe)
{
  // This should only happen if we had invalid input
  PCL_ERROR ("[pcl::%s::performReconstruction2D] Invalid input!\n", getClassName ().c_str ());
  return;
}

// works because all _proj_safe are mutually exclusive
const uint8_t c0 = yz_proj_safe;
const uint8_t c1 = !xy_proj_safe + 1;

for (size_t i = 0, j = 0; i < indices_->size ();++i)
{
  points[j++].input_->points[(*indices_)[i]].data[c0]; //either reserve/push_back or resize/operator[]
  points[j++].input_->points[(*indices_)[i]].data[c1];
}

```